### PR TITLE
Explain why you would want to place a value in the guard

### PR DESCRIPTION
**Pull request by tormol**

### CODE REVIEW

File: src/lib.rs [hunk 1]
```diff
@@ -19,14 +21,12 @@ macro_rules! defer {
 
 /// `Guard` is a scope guard that may own a protected value.
 ///
-/// If you place a guard value in a local variable, its destructor will
+/// If you place a guard in a local variable, the closure will
 /// run regardless how you leave the function — regular return or panic
 /// (barring abnormal incidents like aborts; so as long as destructors run).
 ///
-/// The guard's closure will be called with a mut ref to the held value
-/// in the destructor. It's called only once.
```
Comment (bluss) [inline src/lib.rs]: Can't this sentence remain? Thanks.

Comment (tormol) [inline src/lib.rs]: I have now worked the sentences into the other paragraphs.

I also changed the crate root doc, which I didn't realize was committed, and put it in the readme and crate description.

**Comment by tormol** [comment]
Anything else that should be changed?  
Should I squash?

**Comment by bluss** [comment]
No, it's good. I would want to revisit the wording overall ("even if the code panics") since the situation is much more interesting now that panic runtimes are switchable. In an aborting scenario, scope guard does not get to run on panic.

**Comment by tormol** [comment]
Done.

**Comment by bluss** [comment]
Thank you!

**Pull request merged**